src/index.js

```diff
@@ -88,52 +89,48 @@ const plugin = {
 				},
 			],
 		},
-	},
-};
-
-plugin.configs.recommended = [
-	/** @type {Config & {language:string}} */
-	({
-		name: "markdown/recommended",
-		files: ["**/*.md"],
-		language: "markdown/commonmark",
-		plugins: {
-			markdown: plugin,
-		},
-		rules: /** @type {RulesRecord} */ (recommendedRules),
-	}),
-];
-
-plugin.configs.processor = [
-	{
-		name: "markdown/recommended/plugin",
-		plugins: {
-			markdown: plugin,
-		},
-	},
-	{
-		name: "markdown/recommended/processor",
-		files: ["**/*.md"],
-		processor: "markdown/markdown",
-	},
-	{
-		name: "markdown/recommended/code-blocks",
-		files: ["**/*.md/**"],
-		languageOptions: {
-			parserOptions: {
-				ecmaFeatures: {
-					// Adding a "use strict" directive at the top of
-					// every code block is tedious and distracting, so
-					// opt into strict mode parsing without the
-					// directive.
-					impliedStrict: true,
+		recommended: [
+			{
+				name: "markdown/recommended",
+				files: ["**/*.md"],
+				language: "markdown/commonmark",
+				plugins: (recommendedPlugins = {}),
+				rules: recommendedRules,
+			},
+		],
+		processor: [
+			{
+				name: "markdown/recommended/plugin",
+				plugins: (processorPlugins = {}),
+			},
+			{
+				name: "markdown/recommended/processor",
+				files: ["**/*.md"],
+				processor: "markdown/markdown",
+			},
+			{
+				name: "markdown/recommended/code-blocks",
+				files: ["**/*.md/**"],
+				languageOptions: {
+					parserOptions: {
+						ecmaFeatures: {
+							// Adding a "use strict" directive at the top of
+							// every code block is tedious and distracting, so
+							// opt into strict mode parsing without the
+							// directive.
+							impliedStrict: true,
+						},
+					},
+				},
+				rules: {
+					...processorRulesConfig,
 				},
 			},
-		},
-		rules: {
-			...processorRulesConfig,
-		},
+		],
 	},
-];
+};
+
+// @ts-expect-error
+recommendedPlugins.markdown = processorPlugins.markdown = plugin;
```

Comment:
Using variables instead of `plugin.configs.recommended[0]` and `plugin.configs.processor[0]` so we don't have to care if the order/structure of those config arrays changes.